reasoner/planner/RecursivePlanner.java

```diff
@@ -30,17 +29,14 @@
 import com.vaticle.typedb.core.reasoner.planner.OrderingCoster.LocalSingleCallCosting;
 import com.vaticle.typedb.core.traversal.TraversalEngine;
 
-import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.HashSet;
-import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.Stack;
 
 import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
 
-public class RecursivePlanner extends ReasonerPlanner {
+public abstract class RecursivePlanner extends ReasonerPlanner {
```

Comment:
Ideally, the resolvable ordering search would be a member of this instead of a subclass.
I would need to expose the needed methods / members ( plan / AnswerCountEstimator, OrderingCoster, ConjunctionGraph).